exec/java-exec/src/test/java/org/apache/drill/TestDynamicUDFSupport.java

```diff
@@ -271,6 +271,75 @@ public void testDuplicatedFunctionsInLocalRegistry() throws Exception {
   }
 
   @Test
+  public void testSuccessfulRegistrationAfterSeveralRetryAttempts() throws Exception {
```

Comment:
The case in question would seem to occur when multiple registrations happen concurrently. Do we have a test for that case? That is, do we try to run two thread (two Drillbits?) and have them each attempt to register jars?